src/hooks/useAutoSlippageTolerance.ts

```diff
@@ -14,14 +14,28 @@ import useStablecoinPrice, { useStablecoinValue } from './useStablecoinPrice'
 const V3_SWAP_DEFAULT_SLIPPAGE = new Percent(50, 10_000) // .50%
 const ONE_TENTHS_PERCENT = new Percent(10, 10_000) // .10%
 export const DEFAULT_AUTO_SLIPPAGE = ONE_TENTHS_PERCENT
+const GAS_ESTIMATE_BUFFER = new Percent(10, 100) // 10%
 
 /**
  * Return a guess of the gas cost used in computing slippage tolerance for a given trade
  * @param trade the trade for which to _guess_ the amount of gas it would cost to execute
  */
 function guesstimateGas(trade: Trade<Currency, Currency, TradeType> | undefined): number | undefined {
   if (!!trade) {
-    return 100_000 + trade.swaps.reduce((memo, swap) => swap.route.pools.length + memo, 0) * 30_000
+    let gas = 0
```

Comment:
if this is copied logic, can you link to the location in auto router api as comment?

Reply:
The logic is modified because in the auto router we have additional data to make our guesses based on, but I'll add the link anyway for reference